domain/relation/state/relation_test.go

```diff
@@ -185,20 +289,49 @@ VALUES (?,?,?,0)
 `, applicationEndpointUUID, applicationUUID, charmRelationUUID)
 }
 
-// addCharm inserts a new charm into the database with a predefined UUID, reference name, and architecture ID.
-func (s *relationSuite) addCharm(c *gc.C) {
+// addCharm inserts a new charm into the database with the given UUID.
+func (s *relationSuite) addCharm(c *gc.C, charmUUID string) {
+	// The UUID is also used as the reference_name as there is a unique
+	// constraint on the reference_name, revision and source_id.
 	s.query(c, `
 INSERT INTO charm (uuid, reference_name, architecture_id) 
-VALUES (?, 'app', 0)
-`, fakeCharmUUID)
+VALUES (?, ?, 0)
+`, charmUUID, charmUUID)
 }
 
-// addCharmRelation inserts a new charm relation into the database with the given UUID and predefined attributes.
-func (s *relationSuite) addCharmRelation(c *gc.C, charmRelationUUID string) {
+// addCharmRelationWithDefaults inserts a new charm relation into the database with the given UUID and predefined attributes.
+func (s *relationSuite) addCharmRelationWithDefaults(c *gc.C, charmUUID, charmRelationUUID string) {
 	s.query(c, `
 INSERT INTO charm_relation (uuid, charm_uuid, kind_id, name) 
 VALUES (?, ?, 0, 'fake-provides')
-`, charmRelationUUID, fakeCharmUUID)
+`, charmRelationUUID, charmUUID)
+}
+
+// addCharmRelation inserts a new charm relation into the database with the given UUID and attributes.
+func (s *relationSuite) addCharmRelation(c *gc.C, charmUUID, charmRelationUUID string, r internalcharm.Relation) {
+	s.query(c, `
+INSERT INTO charm_relation (uuid, charm_uuid, kind_id, "key", name, role_id, interface, optional, capacity, scope_id) 
+VALUES (?, ?, 0, ?, ?, ?, ?, ?, ?, ?)
+`, charmRelationUUID, charmUUID, r.Name, r.Name, s.RoleID(r.Role), r.Interface, r.Optional, r.Limit, s.ScopeID(r.Scope))
+}
+
+// RoleID returns the ID used in the database for the given charm role. This
+// reflects the contents of the charm_relation_role table.
+func (s *relationSuite) RoleID(role internalcharm.RelationRole) int {
+	return map[internalcharm.RelationRole]int{
+		internalcharm.RoleProvider: 0,
+		internalcharm.RoleRequirer: 1,
+		internalcharm.RolePeer:     2,
+	}[role]
+}
+
+// ScopeID returns the ID used in the database for the given charm scope. This
+// reflects the contents of the charm_relation_scope table.
+func (s *relationSuite) ScopeID(role internalcharm.RelationScope) int {
```

Comment:
s/func (s *relationSuite) ScopeID/func encodeScopeID/

Keeps with what was done in the application domain.